src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
+
+  catch err
+    error = true
+    callback()
+
   finally
-    global.eval = previousEval
+    if !fn.length and !error
```

Comment:
if there `fn.length is 0` this is a sync call.

basically I want to make two different types using the same signature

```coffee
allowUnsafeEval (callback) ->
# Async
```

```coffee
allowUnsafeEval () ->
# sync
```